From fa96bbba916f57da7d6a7ea01f90b269dff05e59 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 03 Oct 2022 20:16:16 +0300
Subject: [PATCH] [SW-106502] gaudi3: add pcie ctrl fuse function

This patch add function to retrieve pcie ctrl fuse value as it
is required at multiple places.

Change-Id: I5577054d1bf54dbe4603d846522a3dc9465804e0
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 293d916..cd6e246 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -563,6 +563,11 @@
 	return !!(arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SKU_MASK);
 }
 
+int arcboot_rom_fuse_pcie_ctrl_config(void)
+{
+	return !!(arcboot_rom_global.config_fuse & EFUSE_SYSCFG_PCIE_CONTROL_MASK);
+}
+
 void arcboot_rom_halt(void)
 {
 	while (1)
diff --git a/arch/gaudi3/arc/dev_common.h b/arch/gaudi3/arc/dev_common.h
index 9836965..be90e6b 100644
--- a/arch/gaudi3/arc/dev_common.h
+++ b/arch/gaudi3/arc/dev_common.h
@@ -38,3 +38,4 @@
 void arcboot_rom_halt(void);
 bool arcboot_is_valid_start_addr(struct boot_desc *desc);
 int arcboot_rom_efuse_boot_info_skip(void);
+int arcboot_rom_fuse_pcie_ctrl_config(void);
diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index d46c72f..a27ae62 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -130,7 +130,7 @@
 
 static void rom_pcie_recovery(void)
 {
-	uint32_t pcie_ctrl_val = arcboot_rom_global.config_fuse & EFUSE_SYSCFG_PCIE_CONTROL_MASK;
+	uint32_t pcie_ctrl_val = arcboot_rom_fuse_pcie_ctrl_config();
 
 	/*
 	 * Master die (die_id=0): pcie_ctrl 0: PCIe enable 1: PCIe disable
